polyply/tests/test_gen_seq.py

```diff
@@ -48,29 +46,41 @@ def test_branched_graph(branching_f, n_levels, ref_edges):
     assert nx.get_node_attributes(graph, "resname") == resnames
 
 
-
-@pytest.mark.parametrize("macro_str, macro_type, ref_edges",(
-                        ("A:3:1:PEO-1.", "linear", [(0, 1), (1, 2)]),
-                        ("A:3:2:PPI-1.", "branched", [(0, 1), (0, 2), (1, 3), (1, 4), (2, 5), (2, 6)]),
-                        ("A:4:1:PEO-0.5,PPO-0.5","random-linear", [(0, 1), (1, 2), (2, 3)]),
-                        ))
-def test_interpret_macro_string(macro_str, macro_type, ref_edges):
+@pytest.mark.parametrize("macro_str, ref_edges, seeds, resnames", (
+    ("A:3:1:PEO-1.", [(0, 1), (1, 2)], [None, None, None],
+     {0: "PEO", 1: "PEO", 2: "PEO"}),
+    ("A:3:2:PPI-1.",
+     [(0, 1), (0, 2), (1, 3), (1, 4), (2, 5), (2, 6)],
+     [None, None, None, None, None, None, None],
+     {0:"PPI", 1:"PPI", 2:"PPI", 3:"PPI", 4:"PPI", 5:"PPI", 6:"PPI"}),
+    ("A:4:1:PEO-0.5,PPO-0.5",
+     [(0, 1), (1, 2), (2, 3)],
+     [13243, 24940, 98503, 69034],
+     {0:"PPO", 1:"PPO", 2:"PEO", 3:"PPO"}),
+    ("A:4:1:PEO-0.3,PPO-0.2",
+     [(0, 1), (1, 2), (2, 3)],
+     [13243, 24940, 98503, 69034],
+     {0:"PPO", 1:"PPO", 2:"PEO", 3:"PPO"})
+))
+def test_interpret_macro_string(macro_str, ref_edges, seeds, resnames):
     ref_graph = nx.Graph()
     ref_graph.add_edges_from(ref_edges)
```

Comment:
You could consider giving each testcase a single seed, and seeding the RNG as you start the test. That keeps the testing/seeding code out of the actual library.

Also lint

Reply:
see above, this will always just generate a single monomer being selected

Reply:
No, since your RNG will be in a different state after every element you draw from it. If you get a single monomer, that's random (:

Reply:
```pycon
>>> import random
>>> random.seed(0)
>>> for _ in range(10):
...    print(random.choices('ab'))     
['b']
['b']
['a']
['a']
['b']
['a']
['b']
['a']
['a']
['b']
```

Reply:
I see my mistake. Too short of a sequence tried ...